### PR TITLE
Adding r-saga

### PR DESCRIPTION
**Pull request by ocefpaf**
~~Not ready for review yet.~~

See https://cran.r-project.org/web/packages/RSAGA/index.html

### CODE REVIEW

**Comment by ocefpaf** [comment]
@conda-forge/staged-recipes this is ready for review.

File: recipes/r-rsaga/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+{% set posix = 'm2-' if win else '' %}
+{% set native = 'm2w64-' if win else '' %}
+
+package:
+  name: r-rsaga
+  version: {{ version.replace("-", "_") }}
+
+source:
+  fn: RSAGA_{{ version }}.tar.gz
```
Comment (bgruening) [inline recipes/r-rsaga/meta.yaml]: fn can be removed?

Comment (ocefpaf) [reply recipes/r-rsaga/meta.yaml]: Last time I tried this only work for python recipes. Let's try it.

Comment (ocefpaf) [reply recipes/r-rsaga/meta.yaml]: I'm clearly outdated and did not payed attention on the auto-generated shapefile recipe here :smile:

**Comment by jdblischak** [comment]
I realize I'm late here, but a few suggestions for future R recipes:

1. These could have been `noarch` recipes since they do not contain compiled code
1. Please add `conda-forge/r` as a maintainer so that it is easier for us to update in the future

If you use the [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper), these will be done automatically.

**Comment by ocefpaf** [comment]
> If you use the conda_r_skeleton_helper, these will be done automatically.

This should probably be upstreamed.

**Comment by jdblischak** [comment]
> This should probably be upstreamed.

@ocefpaf We upstream what we can. For example, the code to automatically add a license was made much more robust by incorporating it directly into `conda-build` (and we're also working on improving it even more).

The support for noarch is already upstream with the flag `--use-noarch-generic`. It's just not the default setting.

And I don't think we can upstream adding `conda-forge/r` as a maintainer.

**Comment by ocefpaf** [comment]
> And I don't think we can upstream adding conda-forge/r as a maintainer.

That is the only missing part then? I guess we can upstream a `conda skeleton ---maintainer`. The need for two tools is confusing and having everything in skeleton would be nice for all users.

Of course we would still need recommendations to `norach` and what groups should be maintainers, but that is documentation and/or we should add it to the linter. Like, every r recipes add `conda-forge/r`.

**Comment by jdblischak** [comment]
We can already get quite far by recommending the following:

```
conda skeleton cran --use-noarch-generic --add-maintainer conda-forge/r r-shapefiles
```

This adds `noarch` if applicable and adds the maintainter.

> That is the only missing part then? I guess we can upstream a conda skeleton ---maintainer. The need for two tools is confusing and having everything in skeleton would be nice for all users.

I agree it is not ideal. But it is documented right at the top of the documentation for [Contributing packages](https://conda-forge.org/docs/maintainer/adding_pkgs.html#getting-started).

From a maintainer's perspective, there are 2 main advantages to having a dedicated script that we control:

1. We can use the exact formatting we want. For example, we remove all the explanatory comments.
1. We can quickly apply patches. There have been many times where the R recipes start failing unexpectedly, and we need to fix them. The turn around will be a lot longer if we have to wait until our patch is included in the next release of `conda-build`.

@bgruening Thoughts?

**Comment by jakirkham** [comment]
Maybe we should move this to a new issue and try to identify the following.

1. What does `conda skeleton cran` support today?
2. What is missing from `conda skeleton cran`?
3. How does `conda_r_skeleton_helper` address these missing things?
4. What can we do to upstream these things?
5. What remains leftover that we can't upstream?

**Comment by ocefpaf** [comment]
> Maybe we should move this to a new issue and try to identify the following.
> 
> 1. What does `conda skeleton cran` support today?
> 2. What is missing from `conda skeleton cran`?
> 3. How does `conda_r_skeleton_helper` address these missing things?
> 4. What can we do to upstream these things?
> 5. What remains leftover that we can't upstream?

:+1: 

@jdblischak while our new docs are great I really do not expert all users submitting a new recipe to read them. That is why I prefer to focus on tools. We are having lots of discussions and ideas about improving `conda-skeleton` and it would be nice to have more people from the r community in those meetings.

PS0: we had an idea of applying for GSoC and then NumFOCUS Small Grant Development to improve skeleton. Due to lack of time I could not lead that effort but I believe it is still worth it.

PS1: Adding `conda-forge/r` to the linter is tangential to all that and would help the reviewers a lot. This PR was submitted by a core member and approved by another core member familiar with the R standards! Still we both missed that.

**Comment by jdblischak** [comment]
> We are having lots of discussions and ideas about improving conda-skeleton and it would be nice to have more people from the r community in those meetings.

I'm more than happy to participate in any future meetings.

**Comment by ocefpaf** [comment]
> I'm more than happy to participate in any future meetings.

Send me your e-mail so I can add you to the Google calendar. (It can be in PVT, DM e-mail, no need to post it here.)

**Comment by jakirkham** [comment]
Maybe we should add `conda skeleton` as an item for the SciPy sprint? Agree this would be great to work on. @jdblischak, will you be at SciPy?

**Comment by jdblischak** [comment]
> Send me your e-mail so I can add you to the Google calendar. (It can be in PVT, DM e-mail, no need to post it here.)

Done

> @jdblischak, will you be at SciPy?

Unfortunately I can't attend this year

**Comment by jdblischak** [comment]
> Send me your e-mail so I can add you to the Google calendar. 

@ocefpaf Is the meeting happening today? I logged into zoom using the link in the Google Calendar event you invited me to.

**Comment by ocefpaf** [comment]
> @ocefpaf Is the meeting happening today? I logged into zoom using the link in the Google Calendar event you invited me to.

Sorry. Most of us are at SciPy and I guess the remaining core members did not have the meeting.

**Comment by jdblischak** [comment]
@ocefpaf No worries! I'll join the next one in a few weeks. Enjoy SciPy!

**Pull request merged**